cscs-checks/microbenchmarks/gpu/hooks.py

```diff
@@ -0,0 +1,55 @@
+
+import reframe as rfm
+
+
+class SetCompileOpts(rfm.RegressionMixin):
```

Comment:
I would call this hook something different, because - based on the name - I was  expecting compiler cli flags. In this case, it only sets the GPU related modules and set some GPU related cli flags. I think it should be called `SetArchandModules` or something related. What do you think?

Reply:
Another thing that bothers me a bit is having the hooks in separate classes. I'm not sure if it would make sense to stash all of them into the same class.

Reply:
In this case, I vote for what you have implemented. Stashing everything into a single class hinders the code reuse

Reply:
Perhaps we need to think about some mechanism that makes this hook reuse more explicit on the class that uses those hooks. I'm not sure if it would be good to establish a convention where we don't decorate the hooks on base classes. For example, here `SetCompileOpts` would have the member functions undecorated, and then the tests that use these class would explicitly decorate the hooks there ... but I'm not convinced of that approach either 😅